src/lib/protocols/someip.c

```diff
@@ -111,8 +119,8 @@ void ndpi_search_someip (struct ndpi_detection_module_struct *ndpi_struct,
   }
  
   //we extract the Message ID and Request ID and check for special cases later
-  u_int32_t message_id = ntohl(*((u_int32_t *)&packet->payload[0]));
-  u_int32_t request_id = ntohl(*((u_int32_t *)&packet->payload[8]));
+  u_int32_t message_id = ntohl(someip_data_cover_32(&packet->payload[0]));
```

Comment:
#more details:
*   I have just portable the nDpi run my router RT-AC86U version:3.0.0.4.384_81351_koolshare
 
* when I just run the target ,like this :`./ndpiReader -i br0` ,and at the same time,it's broken,notify SIGBUS error occour,so ,there is just no have some_ip.pcap packet
* when the problem occour,i run the cmd `sysinfo `,loged as `Unhandled fault: alignment fault (0x92000021) at 0x00000000f67004aa`

* i don't run do.sh script in my router,because ,it's need envirement;